array_api_compat/cupy/_aliases.py

```diff
@@ -69,20 +67,21 @@
 # asarray also adds the copy keyword, which is not present in numpy 1.0.
 def asarray(
     obj: Union[
-        ndarray,
+        Array,
         bool,
         int,
         float,
-        NestedSequence[bool | int | float],
+        complex,
+        NestedSequence[bool | int | float | complex],
         SupportsBufferProtocol,
     ],
     /,
     *,
-    dtype: Optional[Dtype] = None,
+    dtype: Optional[DType] = None,
     device: Optional[Device] = None,
     copy: Optional[bool] = _copy_default,
```

Comment:
`_copy_default` is not assignable to `bool | None`. Since it's defined as an `object()`, the only way to correctly annotate `copy` is as:

```
    copy: object = _copy_default,
```

I'd suggest using a `enum.Enum` based sentinel here (originally proposed in https://github.com/python/typing/issues/236#issuecomment-227180301), which can properly be annotated:

```py
import enum
from typing import Final, Literal

class _Default(enum.Enum):
    copy = enum.auto()

_CopyDefault: TypeAlias = Literal[_Default.copy]
_copy_default: Final[_CopyDefault] = _Default.copy
```

which can then be used as

```
    copy: bool | _CopyDefault = _copy_default,
```